### DISCUSSION TITLE
Which Ports are used for connection?

### DISCUSSION THREAD

**Discussion by LuDomMedia** [Art-of-WiFi/UniFi-API-client#107] [Q&A] [answered]
Hey, it's me again.

I would like to ask which ports are used for the connection. My problem is, that the Unifi Controller (not UniFi OS based) is behind a firewall and I need to know all the ports.
In the firewall, I've created an entry for the IP of the controller with Port :8443 opened and I still get an 400 Error with a message saying that I have to use TLS with this combination of port and adress.

Could I please get a list of ports? Or is there another error?

**Comment by malle-pietje** [comment]
Hmm, normally that shouldn't happen; where and when exactly are you seeing this error? The API client will normally talk to the controller through the same port as the web interface, reverse proxies can sometimes create issues though.

**Reply by LuDomMedia** [reply]
I'm seeing this error when logging in on the guest authorization portal. The files are more or less the same as in my [last discussion](https://github.com/Art-of-WiFi/UniFi-API-client/discussions/101) and I get the following on the authorized.php:

```
-----------LOGIN-------------
Array
(
    [url] => http://*******************/api/login
    [content_type] => text/plain;charset=UTF-8
    [http_code] => 400
    [header_size] => 76
    [request_size] => 244
    [filetime] => -1
    [ssl_verify_result] => 0
    [redirect_count] => 0
    [total_time] => 0.006029
    [namelookup_time] => 5.0E-6
    [connect_time] => 0.002499
    [pretransfer_time] => 0.00259
    [size_upload] => 47
    [size_download] => 62
    [speed_download] => 10283
    [speed_upload] => 7795
    [download_content_length] => -1
    [upload_content_length] => 47
    [starttransfer_time] => 0.005739
    [redirect_time] => 0
    [redirect_url] => 
    [primary_ip] => ********************
    [certinfo] => Array
        (
        )

    [primary_port] => 8443
    [local_ip] => *******************
    [local_port] => 58420
)

----------RESPONSE-----------
HTTP/1.1 400 
Content-Type: text/plain;charset=UTF-8
Connection: close

Bad Request
This combination of host and port requires TLS.

-----------------------------
Notice: HTTP response status received: 400. Probably a controller login failure in ************/Client.php on line 206
```

**Comment by malle-pietje** [comment, marked as answer]
You should use **https**

**Reply by LuDomMedia** [reply]
Where do I have to change something to make https work?

**Reply by malle-pietje** [reply]
Use https in the URL

**Reply by LuDomMedia** [reply]
Thank you! I'll check if it worked and mark it as done soon.
